src/v/cluster/commands.h

```diff
@@ -154,6 +155,12 @@ using delete_data_policy_cmd = controller_command<
   delete_data_policy_cmd_type,
   model::record_batch_type::data_policy_management_cmd>;
 
+using create_materialized_topic_cmd = controller_command<
+  model::topic_namespace,
+  create_materialized_topic_cmd_data,
```

Comment:
Did this as per advice from Michal, i would assume the pro is to be able to distinguish between the two, since they are of the same type, the only way to know which of the two is the materialized topic_namespace is to look at the source code where the command was created. The con would be this is unnecessary, and there is some wasted space, as we are duplicating a value.